specification/quota/resource-manager/Microsoft.Quota/preview/2019-07-19-preview/quota.json

```diff
@@ -0,0 +1,609 @@
+          "QuotaInformation"
+        ],
+        "summary": "Gets the current quota limit and usages for the resources provider for the specified location.",
+        "description": "Gets the current quota limit and usages for the resources provider for the specified location.",
+        "operationId": "Quota_Check",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ProviderIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Quota information will be returned as response with pagination.",
+            "schema": {
+              "$ref": "#/definitions/CurrentQuotaLimits"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ExceptionResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Quota/providers/{providerId}/locations/{locationId}/quotaRequests/{requestName}": {
+      "put": {
+        "tags": [
+          "QuotaRequests"
+        ],
+        "summary": "Submits a Quota Request for a resource provider at the specified location.",
+        "description": "Submits Quota change request for a resource provider for the specified location. To use, first make a request to the /usages API to get quota information. This information consists of a list of resources and information regarding those resources. For all the resources in that list which require an update to their quotas, update their limit fields in the response from the /usages to their new values. Then, submit this updated JSON object to this quota request API. This will update the quotas to the values specified.",
+        "operationId": "Quota_Request_Submit",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ProviderIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationIdParameter"
+          },
+          {
+            "$ref": "#/parameters/QuotaRequestNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "createQuotaRequests",
+            "in": "body",
+            "description": "Quota requests payload.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CreateGenericQuotaRequestParameters"
+            }
+          }
+        ],
+        "responses": {
+          "202": {
```

Comment:
Since this is an async operation you should add [x-ms-long-running-operation](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation).

Reply:
Sure will update it.
The operation returns immediately with the requestId.
The user can check the status with the GET /subscriptions/{subscriptionId}/providers/Microsoft.Quota/providers/{providerId}/locations/{locationId}/quotaRequests/{requestName}